src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -160,7 +160,8 @@ void ReadFromRemote::addLazyPipe(Pipes & pipes, const ClusterProxy::SelectStream
 
     if (stage == QueryProcessingStage::Complete)
     {
-        add_totals = shard.query->as<ASTSelectQuery &>().group_by_with_totals;
+        if (const auto * ast_select = shard.query->as<ASTSelectQuery>())
```

Comment:
When can it not be an `ASTSelectQuery`?

Reply:
When it is ASTExplainQuery.
https://github.com/ClickHouse/ClickHouse/pull/73994